googlemock/include/gmock/gmock-matchers.h

```diff
@@ -2591,9 +2591,10 @@ class ContainerEqMatcher {
           UniversalPrint(*it, os);
         }
       }
+      return !printed_header2 && !printed_header;
```

Comment:
When containers comparison fails at line 2554, the matcher attempts to compare containers by explicit traversal.
First, the matcher goes over value container and tests the presence of all its' elements in expected containe - lines 2561-2575.  Then the matcher goes over expected container and tests the presence of all its' elements in a value container, in lines 2579-2592. If container don't match, the error is sent to 'os' and correspondent flags are raised. 
If upon these two passes, both print flags are clear, both containers held the same entries's set and matcher should return 'true'. 

Perhaps, better logic would be run these two passes regardless of 'os' stream availability.